File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/web/error/ErrorResponse.java
package com.github.arhor.simple.expense.tracker.web.error;

import java.io.Serializable;
import java.time.temporal.Temporal;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

@JsonPropertyOrder({
    "code",
    "message",
    "details",
    "timestamp",
    "requestId",
})
@JsonInclude(JsonInclude.Include.NON_NULL)
public record ErrorResponse(
    String requestId,
    String message,
    Temporal timestamp,
    ErrorCode code,
    List<String> details
) implements Serializable {
}

File: apps/server/src/main/kotlin/com/github/arhor/simple/expense/tracker/api/error/ErrorResponse.kt
package com.github.arhor.simple.expense.tracker.api.error;

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonPropertyOrder
import java.time.temporal.Temporal

@JsonPropertyOrder(
    "code",
    "message",
    "details",
    "timestamp",
    "requestId",
)
@JsonInclude(JsonInclude.Include.NON_NULL)
data class ErrorResponse(
    val requestId: String,
    val message: String,
    val timestamp: Temporal,
    val code: ErrorCode,
    val details: List<String>,
)
